docs/products/compute/compute-instances/guides/metadata/index.md

```diff
@@ -0,0 +1,218 @@
+
+1.  Deploy a new Compute Instance using your preferred supported distribution image.
+1.  Log in to that instance using SSH or Lish and then modify the cloud-init configuration files (or add your own). These files are typically located in the `/etc/cloud/` folder.
+
+    - `/etc/cloud/cloud.cfg`: The distribution-provided configuration settings.
+    - `/etc/cloud/cloud.cfg.d/`: A directory containing other configuration files that are processed by cloud-init.
+    - `/etc/cloud/cloud.cfg.d/99-linode.cfg`: The Linode-provided cloud-init configuration file. Since this is processed *after* the other configuration files, the settings here override any of the same settings that exist in those other configuration files.
+
+1.  Once you have configured cloud-init with your desired settings, [create a custom image from that Compute Instance](/docs/products/tools/images/guides/capture-an-image/).
+
+Now, when you wish to deploy a new Compute Instance, you can select your custom image. During the creation workflow, you can attach any desired user data for that particular instance. When cloud-init runs, your updated configuration settings will be used alongside any user data that you've added.
+
+## Access the Metadata Service API
+
+In addition to being consumed by cloud-init, the Metadata service can also be accessed through an API. The API is available on industry standard link-local IP addresses (`169.254.169.254` and `fd00:a9fe:a9fe::1`) and returns only instance data and user data for that Compute Instance.
+
+1.  Log in to a Compute Instance that has been deployed in a supported data center using a supported distribution image.
+
+1.  Generate your API token by running the command below:
+
+    ```command
+    curl -X PUT -H "Metadata-Token-Expiry-Seconds: 3600" http://169.254.169.254/v1/token
+    ```
+
+    Instead of receiving the token as an output string, you can save it directly to the `$TOKEN` environmental variable:
+
+    ```command
+    export TOKEN=$(curl -X PUT -H "Metadata-Token-Expiry-Seconds: 3600" http://169.254.169.254/v1/token)
+    ```
+
+1.  Query one of the following API endpoints to receive data from the API. If you did not save the API token to the `$TOKEN` variable, replace `$TOKEN` in the commands below with your token.
+
+    -   **/v1/instance**: Output information about your instance, including plan resources.
+
+        ```command
+        curl -H "Metadata-Token: $TOKEN" http://169.254.169.254/v1/instance
+        ```
+
+    -   **/v1/network**: Output information about your instance's IP addresses.
+
+        ```command
+        curl -H "Metadata-Token: $TOKEN" http://169.254.169.254/v1/network
+        ```
+
+    -   **/v1/user-data**: Output your user data.
+
+        ```command
+        curl -H "Metadata-Token: $TOKEN" http://169.254.169.254/v1/user-data | base64 --decode
```

Comment:
Do we need to provide some kind of security warning for potentially outputting secrets from the config to the user's terminal if they run this command? I'm not familiar with how secrets are generally handled with cloud-init, so maybe this is a non-issue. Maybe this is an opportunity to have a section that describes how secrets are used with cloud-init?

Reply:
I'll log this as feedback for docs supporting the GA release. Thanks!